src/alloc_pool.c

```diff
@@ -108,6 +110,11 @@ static const char *leaking_types[] = {
     "qd_parsed_field_t",   // DISPATCH-1701
     "qdr_delivery_ref_t",  // DISPATCH-1702
 
+#if PN_VERSION_MAJOR == 0 && PN_VERSION_MINOR == 38
```

Comment:
This might mask a leak that @astitcher has been seeing with his proton changes.

Reply:
This is an unfortunate change, agreed. But the I/O layer uncovers a definite leak in the TLS layer in release 0.38.0.

Removing this conditional will cause all the http1 CI tests to fail when built with proton 0.38.

Reply:
If there is a plan to build router container image with patched version of qpid proton 0.38, then it might make sense to create a 0.38.1 branch in proton, put the patches there, and run router CI against that, and not the released 0.38... maybe...

Reply:
I see that this is the only way to have the tests not fail with proton 0.38.0. I am ok with this change.